azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -104,18 +107,31 @@ def serialize(self) -> str:
         return json.dumps(result)
 
     @classmethod
-    def deserialize(cls, problem_as_json: str, name: str):
+    def deserialize(
+            cls, 
+            problem_as_json: str, 
+            name: Optional[str] = None
```

Comment:
This is a breaking change, if I don't pass a `name` here and if the `problem_as_json` doesn't have a name, then the problem will not have a name; previously this scenario would fail because `name` is a required parameter. Should we throw a `ValueError` if `name` is `None` both as input argument and in the problem JSON?

Reply:
Thanks for catching it.
I have handled the None value in the Problem constructor, defaulting it to "Optimization problem".